x-pack/plugins/ml/public/jobs/jobs_list/components/edit_job_flyout/edit_utils.js

```diff
@@ -194,8 +193,14 @@ function extractDetectorDescriptions(job, newJobData) {
 
 function extractCustomSettings(job, newJobData) {
   const settingsData = {};
-  if (job.custom_settings && newJobData && newJobData.customUrls) {
+  if (job.custom_settings !== undefined) {
     settingsData.custom_settings = job.custom_settings;
+  }
+
+  if (newJobData && newJobData.customUrls) {
+    if (settingsData.custom_settings === undefined) {
+      settingsData.custom_settings = {};
```

Comment:
rather than having the previous check for `job.custom_setting` and then this check for `settingsData.custom_settings`  they could be replaced with a single check here:
```
settingsData.custom_settings = job.custom_settings || {};
```
or 
```
settingsData.custom_settings = (job.custom_settings !== undefined) ? job.custom_settings : {};
```